File: PI_PERSO/WEB_SYMFONY/Reclamations/MobilePush/integration --finale--/WebTemplate/src/Entity/Reclamation.php
<?php

namespace App\Entity;

use App\Repository\ReclamationRepository;
use Doctrine\DBAL\Types\Types;
use Symfony\Component\Validator\Constraints as Assert;
use Doctrine\ORM\Mapping as ORM;

/**
 * Reclamation
 *
 * @ORM\Table(name="reclamation", indexes={@ORM\Index(name="fk_userid_rec", columns={"user_id"})})
 * @ORM\Entity(repositoryClass = "App\Repository\ReclamationRepository")
 */
class Reclamation
{
    /**
     * @var int
     *
     * @ORM\Column(name="rec_id", type="integer", nullable=false)
     * @ORM\Id
     * @ORM\GeneratedValue(strategy="IDENTITY")
     */
    private $recId;

    /**
     * @var string
     *
     * @ORM\Column(name="titre_rec", type="string", length=255, nullable=false)
     * @Assert\NotBlank(message="Le titre de la réclamation ne doit pas être vide")
     *@Assert\Length(
    *      min = 2,
    *      minMessage = "La titre doit contenir au moins {{ limit }} caractères."
    * )
     */
    private $titreRec;

    /**
     * @var string
     *
     * @ORM\Column(name="type", type="string", length=255, nullable=false)
     */
    private $type;

    /**
     * @var string
     *
     * @ORM\Column(name="description", type="text", length=65535, nullable=false)
     * @Assert\NotBlank(message="Assurez-vous d'écrire une description pour votre réclamation")
     *@Assert\Length(
    *      min = 10,
    *      minMessage = "La Description doit contenir au moins {{ limit }} caractères."
    * )
     */
    private $description;

    /**
     * @var \DateTime
     *
     * @ORM\Column(name="dateCreation", type="date", nullable=false)
     */
    private $dateCreation;

    /**
     * @var \DateTime|null
     *
     * @ORM\Column(name="dateFin", type="date", nullable=true, options={"default"="NULL"})
     */
    private $dateFin = null;

    /**
     * @var string
     *
     * @ORM\Column(name="status", type="string", length=255, nullable=false, options={"default"="'Ouvert'"})
     */
    private $status = 'Ouvert';

    /**
     * @var \User
     *
     * @ORM\ManyToOne(targetEntity="User")
     * @ORM\JoinColumns({
     *   @ORM\JoinColumn(name="user_id", referencedColumnName="id_user")
     * })
     */
    private $user;

    public function getRecId(): ?int
    {
        return $this->recId;
    }

    public function getTitreRec(): ?string
    {
        return $this->titreRec;
    }

    public function setTitreRec(string $titreRec): self
    {
        $this->titreRec = $titreRec;

        return $this;
    }

    public function getType(): ?string
    {
        return $this->type;
    }

    public function setType(string $type): self
    {
        $this->type = $type;

        return $this;
    }

    public function getDescription(): ?string
    {
        return $this->description;
    }

    public function setDescription(string $description): self
    {
        $this->description = $description;

        return $this;
    }

    public function getDateCreation(): ?\DateTimeInterface
    {
        return $this->dateCreation;
    }

    public function setDateCreation(\DateTimeInterface $dateCreation): self
    {
        $this->dateCreation = $dateCreation;

        return $this;
    }

    public function getDateFin(): ?\DateTimeInterface
    {
        return $this->dateFin;
    }

    public function setDateFin(?\DateTimeInterface $dateFin): self
    {
        $this->dateFin = $dateFin;

        return $this;
    }

    public function getStatus(): ?string
    {
        return $this->status;
    }

    public function setStatus(string $status): self
    {
        $this->status = $status;

        return $this;
    }

    public function getUser(): ?User
    {
        return $this->user;
    }

    public function setUser(?User $user): self
    {
        $this->user = $user;

        return $this;
    }


}

File: PI_Integrated/Mobile Final Version/src/com/mycompany/myapp/entities/Reclamation.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package com.mycompany.myapp.entities;
import java.util.Date; 

/**
 *
 * @author Theto
 */

public class Reclamation {
    private int rec_id;
    private int user_id;
    private String titre_rec;
    private String type;
    private String description;
    private Date date_creation;
    private Date date_fin;
    private String status = "Ouvert";

    public Reclamation() {
    }

    public Reclamation(int user_id, String titre_rec, String type, String description, String status) {
        this.user_id = user_id;
        this.titre_rec = titre_rec;
        this.type = type;
        this.description = description;
        Date date_deb = new Date();
        this.date_creation = date_deb;
        this.status = status;
    }
        public Reclamation(int rec_id, int user_id, String titre_rec, String type, String description, String status) {
        this.rec_id = rec_id;
        this.user_id = user_id;
        this.titre_rec = titre_rec;
        this.type = type;
        this.description = description;
        Date date_deb = new Date();
        this.date_creation = date_deb;
        this.status = status;
    }

    public Reclamation(int i) {
    }

    public Reclamation(int rec_id, int user_id, String titre_rec, String type, String description, String status, Date date_creation, Date date_fin) {
        this.rec_id = rec_id;
        this.user_id = user_id;
        this.titre_rec = titre_rec;
        this.type = type;
        this.description = description;
        this.date_creation = date_creation;
        this.date_fin = date_fin;
        this.status = status;
    }

    public int getRec_id() {
        return rec_id;
    }

    public void setRec_id(int rec_id) {
        this.rec_id = rec_id;
    }

    public int getUser_id() {
        return user_id;
    }

    public void setUser_id(int user_id) {
        this.user_id = user_id;
    }

    public String getTitre_rec() {
        return titre_rec;
    }

    public void setTitre_rec(String titre_rec) {
        this.titre_rec = titre_rec;
    }

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public Date getDate_creation() {
        return date_creation;
    }

    public void setDate_creation(Date date_creation) {
        this.date_creation = date_creation;
    }

    public Date getDate_fin() {
        return date_fin;
    }

    public void setDate_fin(Date date_fin) {
        this.date_fin = date_fin;
    }

    public String getStatus() {
        return status;
    }

    public void setStatus(String status) {
        this.status = status;
    }

    @Override
    public String toString() {
        return "Reclamation{" + "rec_id=" + rec_id + ", user_id=" + user_id + ", titre_rec=" + titre_rec + ", type=" + type + ", description=" + description + ", date_creation=" + date_creation + ", date_fin=" + date_fin + ", status=" + status + '}';
    }
    
    
    
    
    
    
    
    
}
